specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json

```diff
@@ -0,0 +1,421 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The list of configuration profile assignment intents were returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntentList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    },
+    "/providers/Microsoft.Automanage/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "Lists all of the available Automanage REST API operations.",
+        "x-ms-examples": {
+          "Lists all of the available Automanage REST API operations": {
+            "$ref": "./examples/listOperations.json"
+          }
+        },
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OperationList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "ConfigurationProfileAssignmentIntentList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+          },
+          "description": "Result of the list configuration profile assignment intent operation."
+        }
+      },
+      "description": "The response of the list configuration profile assignment intent operation."
+    },
+    "ConfigurationProfileAssignmentIntent": {
```

Comment:
This is almost identical to configurationProfileAssignments.  What is the difference and why do you need both?

Reply:
Thank you Chris for noticing and asking question!

1) configurationProfileAssignments has the property **"x-ms-long-running-operation": true**. It is long running operation. So it has the provisioningState as per the ARM guidelines.

2) configurationProfileAssignmentIntents has the property **"x-ms-long-running-operation": false**. It is not long running. When customers call the PUT configurationProfileAssignmentIntents, it do a PUT on configurationProfileAssignments  and return immediately.

We are integrating Automanage with CreateVM flow in portal UX. When we use configurationProfileAssignments it takes about 10+ min to reach a terminal state and deployment waits for the terminal state. As per the compute portal team expectation or standards they cannot wait for that long. 

**Once we complete the implementation of configurationProfileAssignmentIntents we will use that in CreateVM flow in portal UX. So we complete the operation in 4 sec or less.**

Reply:
Related post: https://stackoverflow.microsoft.com/questions/248238

Reply:
How does a user know that the provisioning is complete? Say that a user calls your API, how can they poll to get the provisioningState? 

---
In reply to: [603703789](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603703789) [](ancestors = 603703789)

Reply:
Thank you!

@jorgecotillo, I assume you are asking the above question for configurationProfileAssignmentIntents.

The configurationProfileAssignmentIntents resource will do synchronous operation and it will not have the provisioningState property.  If there is no provisioningState then ARM will assume as succeeded and it will not poll.

Reply:
Yeah, I think I get that part, but if I am a user, and I make an api call to this intent endpoint, how do I know if configurationProfileAssignment provisioning succeeded/failed? This intent seems like a fire and forget type of situation where returns fast and internally does a PUT on configurationProfileAssignment, I think you'll need to document this pretty well, is that your plan to avoid customer confusion?

---
In reply to: [605302690](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r605302690) [](ancestors = 605302690)

Reply:
Thank you Jorge! It is valid question.

We have two scenarios

1. **SDK/API** : We need to add clear documentation for this as you have mentioned. Thank you for pointing this out. To add documentation opened  [**Task 9651989: [Doc]ConfigurationProfileAssignmentIntents**](https://msazure.visualstudio.com/One/_workitems/edit/9651989)
2.  **Portal** : When we consume the configurationProfileAssignmentIntents PUT operation in create VM flow. The customers will navigate to VM  blade and we have Automanage (in table of contents) which already fetches the configurationProfileAssignment . So portal should be good.